pkg/sql/logictest/testdata/logic_test/set_schema

```diff
@@ -0,0 +1,205 @@
+statement ok
+SET experimental_enable_user_defined_schemas = true
+
+statement ok
+CREATE SCHEMA s1
+
+statement ok
+CREATE SCHEMA s2
+
+statement ok
+CREATE TABLE t(x INT)
+
+# No error should happen if IF EXISTS is used and the table does not exist.
+statement ok
+ALTER TABLE IF EXISTS does_not_exist SET SCHEMA s2
+
+# Cannot set schema to one that does not exist.
+statement error pq: schema does_not_exist does not exist
+ALTER TABLE t SET SCHEMA does_not_exist
+
+# Ensure we cannot set schema if there is already a table in the schema
```

Comment:
nit no period at the end here

also the comment should be "if there is a table with the same name in the schema"